crates/ruff_linter/src/rules/ruff/rules/non_empty_init_module.rs

```diff
@@ -0,0 +1,139 @@
+use ruff_macros::{ViolationMetadata, derive_message_formats};
+use ruff_python_ast::{self as ast, Stmt};
+use ruff_python_semantic::analyze::typing::is_type_checking_block;
+use ruff_text_size::Ranged;
+
+use crate::{Violation, checkers::ast::Checker};
+
+/// ## What it does
+///
+/// Detects the presence of code in `__init__.py` files.
+///
+/// ## Why is this bad?
+///
+/// Including code in `__init__.py` files can cause surprising slowdowns because the code is run at
+/// import time. `__init__.py` files are also often empty, so it's easy to overlook them when
+/// debugging these issues.
+///
+/// ## Example
+///
+/// ```python
+/// """My module docstring."""
+///
+///
+/// class MyClass:
+///     def my_method(self): ...
+/// ```
+///
+/// Use instead:
+///
+/// ```python
+/// """My module docstring."""
+///
+/// from submodule import MyClass
+///
+/// __all__ = ["MyClass"]
+/// ```
+///
+/// When [`lint.ruff.strictly-empty-init-modules`] is enabled, all code, including imports, will be
+/// flagged:
+///
+/// ```python
+/// from submodule import MyClass
+///
+/// __all__ = ["MyClass"]
+/// ```
+///
+/// And the only fix is entirely emptying the file:
+///
+/// ```python
+/// ```
+///
+/// ## Options
+/// - [`lint.ruff.strictly-empty-init-modules`]
+///
+/// ## References
+/// - [`flake8-empty-init-modules`](https://github.com/samueljsb/flake8-empty-init-modules/)
+#[derive(ViolationMetadata)]
+#[violation_metadata(preview_since = "0.14.11")]
+pub(crate) struct NonEmptyInitModule;
+
+impl Violation for NonEmptyInitModule {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        "Code detected".to_string()
+    }
+}
+
+/// RUF070
+pub(crate) fn non_empty_init_module(checker: &Checker, stmt: &Stmt) {
+    if !checker.in_init_module() {
+        return;
+    }
+
+    let semantic = checker.semantic();
+
+    // Only flag top-level statements
+    if semantic.current_statement_parent().is_some() {
+        return;
+    }
+
+    if !checker.settings().ruff.strictly_empty_init_modules {
+        if semantic.in_pep_257_docstring() || semantic.in_attribute_docstring() {
```

Comment:
Can you say more about the reason for allowing attribute docstrings, given that attributes are denied by default?

Reply:
Not having to use two suppressions seems desirable (for as long as it doesn't result in any false negatives)